# Bonus: Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt

In [2]:
# "tobs" is "temperature observations"
df = pd.read_csv('data/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [3]:
# Convert the date column format from string to datetime
df['date'] = df['date'].astype('datetime64')

### Compare June and December data across all years 

In [4]:
from scipy import stats

In [5]:
# Filter data for desired months
june_df = df[df['date'].dt.month == 6]
december_df = df[df['date'].dt.month == 12]

In [6]:
# Identify the average temperature for June
june_avg_temp = round(june_df['tobs'].mean(), 2)
june_avg_temp

74.94

In [7]:
# Identify the average temperature for December
december_temp = round(december_df['tobs'].mean(), 2)
december_temp

71.04

In [8]:
# Run paired t-test
stats.ttest_ind(june_df.tobs, december_df.tobs, equal_var=False)

Ttest_indResult(statistic=31.355036920962423, pvalue=4.193529835915755e-187)

### Analysis

In this scenario we use a paired t-test, the reason being that our objective is to determine if there is a significant difference between the means of two dependent samples. And we will not assume equal variance between groups.

Having run the paired t-test, we fail to reject the null hypothesis that the mean temperatures in Hawaii in June vs. in December are the same, because we observe a fairly large p-value of ~4.2. Thus, at the α=0.5 (p-value of 4.2 > α of 0.5) level of significance, there is not sufficient evidence to conclude that there is a meaningful difference between the temperature in June and December in Hawaii.
